jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -1424,6 +1424,10 @@ ecma_bytecode_deref (ecma_compiled_code_t *bytecode_p) /**< byte code pointer */
       return;
     }
 #endif /* JERRY_DEBUGGER */
+
+#ifdef JMEM_STATS
+    jmem_stats_free_byte_code_bytes (((size_t) bytecode_p->size) << JMEM_ALIGNMENT_LOG);
+#endif /* JMEM_STATS */
```

Comment:
Yes, because it measures the parser byte code only, regex byte code is different topic. I don't think regexes consumes that much memory in most cases.